exercises/diffie-hellman/canonical-data.json

```diff
@@ -0,0 +1,67 @@
+{
+  "exercise": "diffie-hellman",
+  "version": "1.0.0",
+  "comments": [
+    "Optional tests to consider:",
+    "* Validation of parameters:",
+    "  Validate that `p`, `g` are valid.",
+    "  Validate that keys given as inputs are valid.",
+    "  Resources that show what happens if parameters are not validated:",
+    "  http://cryptopals.com/sets/5/challenges/34",
+    "  http://cryptopals.com/sets/5/challenges/35",
+    "* Large numbers:",
+    "  Although the calculations fundamentally do not require large numbers,",
+    "  this is a reasonable real-world use for them",
+    "  and it may be instructive to have an exercise on their use.",
+    "  Consult tracks with this exercise (such as the Go track) for possible inputs to use."
+  ],
+  "cases": [
+    {
+      "description": "private key is in range 1 .. p",
+      "property": "privateKeyIsInRange",
```

Comment:
Typically `property` is used as a function name... perhaps this should simply be `"privateKey"`?

Reply:
> Typically `property` is used as a function name... perhaps this should simply be `"privateKey"`?

In the following discussion, I will use the https://github.com/exercism/problem-specifications/blob/master/README.md 's definition of `property`, which to define the type of test. Further, I acknowledge that the README agrees with what you say: Typically, it is the function name.

Here, I think it made sense to use the broader definition of simply the type of test. Specifically, I say that means this: If two tests have the same `property`, then necessarily their inputs and outputs should be interpreted in the same way.

Let's brainstorm some suggestions here. Should it be the case that *both* `privateKeyIsRandom` and `privateKeyIsInRange` be changed to just `privateKey`?

If so, I think we will have to rely heavily on `expected` to express the difference between the two. This is, of course, very possible. I'm happy to go with anything suggested, since there is no precedent.

Reply:
> Let's brainstorm some suggestions here. Should it be the case that both privateKeyIsRandom and privateKeyIsInRange be changed to just privateKey?

This would be my preference too (same as @cmccandless).

> If so, I think we will have to rely heavily on expected to express the difference between the two. This is, of course, very possible. I'm happy to go with anything suggested, since there is no precedent.

I think using the expected value to discern between what is being tested for the private key is the best option here.

Reply:
> If so, I think we will have to rely heavily on expected to express the difference between the two. This is, of course, very possible. I'm happy to go with anything suggested, since there is no precedent.

I am having second thoughts about using `expected` for this. I will have to write a comment later explaining why I have second thoughts.